catalog/core/catalog-core-commons/src/main/java/ddf/util/XPathHelper.java

```diff
@@ -250,14 +258,38 @@ public synchronized Object evaluate(String xpathExpressionKey, QName returnType,
                 .setNamespaceContext(nsContext);
 
         XPathExpression compiledExpression = XPathCache.getCompiledExpression(xpathExpressionKey);
-
         Thread thread = Thread.currentThread();
         ClassLoader loader = thread.getContextClassLoader();
         thread.setContextClassLoader(this.getClass()
                 .getClassLoader());
 
+        DocumentBuilderFactory df =DocumentBuilderFactory.newInstance();
+        df.setNamespaceAware(true);
+        try {
+            df.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+            df.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        } catch (ParserConfigurationException e) {
+            throw new XPathExpressionException(e);
+        }
+        DocumentBuilder documentBuilder;
+        byte[] array;
+        try {
+            documentBuilder = df.newDocumentBuilder();
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(document);
+            ByteArrayOutputStream bos=new ByteArrayOutputStream();
```

Comment:
Even though it's a no-op on a BAOS, should probably declare this in a try-with-resources statement or it will result in a false positive warning from static analysis tools.